### PR TITLE
Add Support for Fetching Remote Catalog

### PR DESCRIPTION
**Pull request by theishshah**
Contains implementation changes, needs utest and documentation updates.

### CODE REVIEW

File: cmd/opm/alpha/template/composite.go [hunk 1]
```diff
@@ -31,11 +34,23 @@ and a 'composite template' file`,
 		Args: cobra.MaximumNArgs(0),
 		Run: func(cmd *cobra.Command, args []string) {
 			containerTool = "docker"
-			catalogData, err := os.Open(catalogFile)
+			var tempCatalog io.ReadCloser
```
Comment (grokspawn) [inline cmd/opm/alpha/template/composite.go:37]: Probably a stupid question w.r.t. golang and deferred calls, but is there any reason that we couldn't do 
```golang
var catalogData io.ReadCloser
```
here, and in the conditionals, assign to it/defer closure to it as needed? 

If it's imposed by the language, fine, but L53 feels like we could optimize it away with this approach.

File: cmd/opm/alpha/template/composite.go [hunk 2]
```diff
@@ -31,11 +34,23 @@ and a 'composite template' file`,
...
-				log.Fatalf("opening catalog config file %q: %s", catalogFile, err)
+				tempCatalog, err = os.Open(catalogFile)
+				if err != nil {
+					log.Fatalf("opening catalog config file %q: %s", catalogFile, err)
+				}
+				defer tempCatalog.Close()
+			} else {
+				tempResp, err := http.Get(catalogURI.String())
+				if err != nil {
+					log.Fatalf("fetching remote catalog config file %q: %s", catalogFile, err)
```
Comment (grokspawn) [inline cmd/opm/alpha/template/composite.go:48]: Unrelated to this PR, per se, but one of the confusing bits of nomenclature we have in this space is the names that we use to designate the spec/files from the operator author and catalog maintainer which fulfill the roles of this conversation.  `catalog/config` and `catalogs.yaml` are too similar.  
We're effectively canonicalizing `catalog config file` here, which I think is fine, so maybe we start referring to the operator author's file as `contributions spec/file` WDYT?

Comment (everettraven) [reply cmd/opm/alpha/template/composite.go:48]: That sounds reasonable to me

Comment (theishshah) [reply cmd/opm/alpha/template/composite.go]: /resolve

File: cmd/opm/alpha/template/composite.go (hunk 1 above)
Comment (theishshah) [reply cmd/opm/alpha/template/composite.go:37]: /resolve

**Review by everettraven — APPROVED** [review]
/lgtm

**Pull request merged**